great_expectations/datasource/fluent/sql_datasource.py

```diff
@@ -797,7 +797,7 @@ class TableAsset(_SQLAsset):
 
     # Instance fields
     type: Literal["table"] = "table"
-    table_name: str
+    table_name: str = ""
```

Comment:
`non-blocking`
```suggestion
    table_name: str = Field("", description="Name of SQL table. Will default to the value of `name` if not provided.")
```
I would use a `pydantic.Field()` here so we can provide some extra metadata to the json schema.
You'll have to run `invoke schema --sync` again after this.